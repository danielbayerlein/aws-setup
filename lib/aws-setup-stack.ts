import * as cdk from '@aws-cdk/core'
import { BillingAlert } from '@moralesl/billing-alert'
import 'dotenv/config'

export class AwsSetupStack extends cdk.Stack {
  constructor (scope: cdk.App, id: string, props?: cdk.StackProps) {
    super(scope, id, props)

    new BillingAlert(this, 'BillingAlert', {
      amount: 50,
      emails: [process.env.EMAIL!]
    })
  }
}
