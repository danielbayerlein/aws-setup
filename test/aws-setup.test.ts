import { SynthUtils, expect as expectCDK, haveResource } from '@aws-cdk/assert'
import * as cdk from '@aws-cdk/core'
import * as AwsSetup from '../lib/aws-setup-stack'

describe('AwsSetupStack', () => {
  const EMAIL = 'test@example.com'

  beforeAll(() => {
    process.env.EMAIL = EMAIL
  })

  test('matches snapshot', () => {
    const app = new cdk.App()
    const stack = new AwsSetup.AwsSetupStack(app, 'MyTestStack')
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot()
  })

  test('BillingAlert', () => {
    const app = new cdk.App()
    const stack = new AwsSetup.AwsSetupStack(app, 'MyTestStack')
    const AMOUNT = 50

    expectCDK(stack).to(
      haveResource('AWS::SNS::Subscription', {
        Endpoint: EMAIL
      })
    )

    expectCDK(stack).to(
      haveResource('AWS::CloudWatch::Alarm', {
        Threshold: AMOUNT
      })
    )
  })
})
