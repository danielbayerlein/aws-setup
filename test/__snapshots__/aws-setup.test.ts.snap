// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AwsSetupStack matches snapshot 1`] = `
Object {
  "Resources": Object {
    "BillingAlarm50Dollars0F34B408": Object {
      "Properties": Object {
        "AlarmActions": Array [
          Object {
            "Ref": "BillingAlert50DollarsTopicEDF128F0",
          },
        ],
        "AlarmDescription": "Account Billing Alert for $50",
        "AlarmName": "Billing Alert - Estimated Bill Exceeds $50",
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "Dimensions": Array [
          Object {
            "Name": "Currency",
            "Value": "USD",
          },
        ],
        "EvaluationPeriods": 1,
        "MetricName": "EstimatedCharges",
        "Namespace": "AWS/Billing",
        "Period": 21600,
        "Statistic": "Maximum",
        "Threshold": 50,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "BillingAlert50DollarsTopicEDF128F0": Object {
      "Properties": Object {
        "TopicName": "Billing-Alert-50Dollars",
      },
      "Type": "AWS::SNS::Topic",
    },
    "BillingAlert50DollarsTopictestexamplecom0100BC4A": Object {
      "Properties": Object {
        "Endpoint": "test@example.com",
        "Protocol": "email",
        "TopicArn": Object {
          "Ref": "BillingAlert50DollarsTopicEDF128F0",
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
  },
}
`;
